backend/modules/database/db_queries.bal

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2025 WSO2 LLC. (https://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+import ballerina/sql;
+
+# [Database] Build query to fetch apps visible to any of the given user roles.
+# Apps with empty user_groups are visible to everyone.
+#
+# + roles - User roles to filter
+# + return - Parameterized query selecting apps for the roles
+isolated function fetchAppsByUserRolesQuery(string[] roles) returns sql:ParameterizedQuery {
+    sql:ParameterizedQuery selectClause = `
+        SELECT 
+            id,
+            header,
+            url,
+            description,
+            version_name,
+            tag_id,
+            icon,
+            added_by
+        FROM apps
+        WHERE is_active = 1`;
+
+    if roles.length() == 0 {
+        // No roles provided: show only apps with empty user_groups
+        return sql:queryConcat(selectClause, `
+          AND (user_groups IS NULL OR user_groups = '')
+        ORDER BY header`);
+    }
+
+    // Roles provided: show apps matching any role OR with empty user_groups
+    sql:ParameterizedQuery[] roleConditions = [];
+    foreach var role in roles {
+        roleConditions.push(`FIND_IN_SET(${role}, user_groups)`);
+    }
```

Comment:
Can't we use [sql:arrayFlattenQuery](https://central.ballerina.io/ballerina/sql/latest#arrayFlattenQuery) in here